security/grub-extend-pcr.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2023 IBM
+# Author: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>
+
+import os
+from avocado import Test
+from avocado.utils import distro, genio, process
+from avocado.utils.software_manager.manager import SoftwareManager
+
+
+class GrubExtendPCR(Test):
+    """
+    Test case for testing grub extend PCR
+    :avocado: tags=security,grub,tpm,pcr
+    """
+
+    def setUp(self):
+        '''
+        Install the basic packages
+        '''
+        # Check for basic utilities
+        val = genio.read_file("/proc/cpuinfo")
+        if 'POWER10' not in val:
+            self.cancel("Power10 LPAR is required to run this test.")
+        device_tree_path = "/proc/device-tree/vdevice/"
+        d_list = os.listdir(device_tree_path)
+        vtpm = [i for i, item in enumerate(d_list) if item.startswith('vtpm@')]
+        if not vtpm:
+            self.cancel("vTPM not enabled.")
+        smm = SoftwareManager()
+        deps = []
+        detected_distro = distro.detect()
+        if detected_distro.name in ['rhel', 'fedora', 'centos', 'redhat']:
+            deps.extend(["tss2"])
+        elif 'SuSE' in detected_distro.name:
+            deps.extend(['ibmtss'])
+        else:
+            self.cancel("%s not supported for this test" %
+                        detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+
+    def _run_pcrread(self, cmd):
+        '''
+        Run tsspcrread command, validate output
+        '''
+        output = process.system_output(cmd, ignore_status=True).decode().splitlines()
+        if '00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00' in output:
+            self.fail("%s Failed to give correct output." % cmd)
+        '''
+        Fail case:
+        count 1 pcrUpdateCounter 39
+        digest length 32
+        00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
+        '''
```

Comment:
Please move comments inside the method doc string

Reply:
Resolved.